src/test/java/com/google/api/gax/grpc/RecordingScheduler.java

```diff
@@ -29,41 +29,63 @@
  */
 package com.google.api.gax.grpc;
 
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Mockito.when;
+
 import java.util.ArrayList;
 import java.util.List;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.ScheduledFuture;
 import java.util.concurrent.ScheduledThreadPoolExecutor;
 import java.util.concurrent.TimeUnit;
 import org.joda.time.Duration;
+import org.mockito.Mockito;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
 
-class RecordingScheduler implements UnaryCallable.Scheduler {
-  private final ScheduledExecutorService executor;
-  private final List<Duration> sleepDurations = new ArrayList<>();
-  private final FakeNanoClock clock;
+abstract class RecordingScheduler implements ScheduledExecutorService {
```

Comment:
Because otherwise I would have to implement 17 other methods in ScheduledExecutorService interface, which are irrelevant for our test. I refactored this class, it used to be implementing UnaryCallable.Scheduler instead, which was a wrapper interface. I believe that interface existed only because of the problems with mocking the full scheduler for the testing purposes. 

Now, when I think about it, it does not even have to be a class (can be just an interface, since the implementation is still completely generated  Mockito during runtime). I will change it to that, if it is ok for you.